arrows/pytorch/SRNN_matching.py

```diff
@@ -123,8 +123,9 @@ def __call__(self, track_set, track_state_list, track_search_threshold):
             TargetRNNDataLoader(track_set, track_state_list, track_search_threshold, RnnType.Target_RNN_AIM),
             batch_size=self._batch_size, shuffle=False, **kwargs)
 
-        self._est_similarity(AIM_V_data_loader, RnnType.Target_RNN_AIM_V)
-        self._est_similarity(AIM_data_loader, RnnType.Target_RNN_AIM)
+        with torch.no_grad():
```

Comment:
@neal-siekierski I did not understand your point here. Why do we need to re-enable gradient? Since we only do the forward pass, we do not need any gradient information.

Reply:
Gradient computation is enabled by default, and none of the code that I've seen explicitly enables it. Moreover, the emphasis in the documentation is on *locally* disabling gradient computation: https://pytorch.org/docs/stable/autograd.html#locally-disabling-gradient-computation. While I don't think it matters for my current use-case, where indeed only the forward pass is needed, potential future code elsewhere that tries to do training after this method has run would otherwise fail because gradient computation is disabled.

Reply:
@neal-siekierski For this tracking code, we only allow forward pass. Not sure we really would have the requirements that training during tracking. At this moment, the training part is a separate project. From my point of view, when we deploy a model, we need to disable gradient in order to have a better performance in terms of forwarding speed.

Reply:
The `with`-block makes sure that gradient calculation is disabled when running inference in `_est_similarity`. The only difference behavior-wise is that, once inference is done, the flag that indicates whether to do gradient calculation is restored to its previous state. So that still does what we're looking for performance-wise, right?

(Also, looking at the docs again, I see that `torch.no_grad()` can also be used as a decorator. Using it on the definition of `_est_similarity` directly rather than in a `with`-block elsewhere would be clearer than what I have currently, I think.)